gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
+        inlier_cameras: Set[int] = set()
+        for (i1, i2) in w_i2Ui1_dict:
+            if (C(i2), C(i1)) in inliers:  # there is a flip in indices from w_i2Ui1_dict to inliers.
+                inlier_w_i2Ui1_dict[(i1, i2)] = w_i2Ui1_dict[(i1, i2)]
+                inlier_cameras.add(i1)
+                inlier_cameras.add(i2)
+
+        for (track_id, cam_id) in w_i2Ui1_dict_tracks:
+            # Same as above, `inliers` contains symbols that are flipped - C(cam_id), L(track_id).
+            # Only add an inlier camera-track measurements if the camera has other camera-camera inliers.
+            if (C(cam_id), L(track_id)) in inliers and cam_id in inlier_cameras:
+                inlier_w_i2Ui1_dict_tracks[(track_id, cam_id)] = w_i2Ui1_dict_tracks[(track_id, cam_id)]
+
+        return inlier_w_i2Ui1_dict, inlier_w_i2Ui1_dict_tracks, inlier_cameras
+
+    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]) -> gtsam.Values:
         """Converts translations from a list of absolute poses to gtsam.Values for initialization.
 
         Args:
-            wTi_initial: List of
+            wTi_initial: List of absolute poses.
+
+        Returns:
+            gtsam.Values containing initial translations (uses symbols for camera index).
         """
         initial = gtsam.Values()
         for i, wTi in enumerate(wTi_initial):
             if wTi is not None:
-                initial.insertPoint3(i, wTi.value.translation())
+                initial.insertPoint3(C(i), wTi.value.translation())
         return initial
 
+    def _select_tracks_for_averaging_by_measurements(
+        self,
+        tracks: List[SfmTrack2d],
+        valid_cameras: Set[int],
+        intrinsics: List[Optional[gtsfm_types.CALIBRATION_TYPE]],
+        measurements_per_camera=TRACKS_MEASUREMENTS_PER_CAMERA,
+    ) -> List[SfmTrack2d]:
+        """Removes bad tracks and selects the longest ones until all cameras see `measurements_per_camera` tracks.
+
+        Bad tracks are those that have fewer than 3 measurements from valid_cameras.
+        Sorts the remaining tracks in descending order, then selects the longest ones until all cameras see at least
+         `measurements_per_camera` tracks.
+        This is based on 1dsfm's implementation here:
+        https://github.com/wilsonkl/SfM_Init/blob/fd012ef93462b8623e8d65fa0c6fa95b32270a3c/sfminit/transproblem.py#L235
+
+        Args:
+            tracks: List of all input tracks.
+            valid_cameras: Set of valid camera indices (these have direction measurements and valid rotations).
+            intrinsics: List of camera intrinsics.
+            measurements_per_camera: Number of track direction measurements that need to be observed by each camera.
+
+        Returns:
+            List of tracks to use for averaging.
+        """
+        filtered_tracks = []
+        valid_cameras_with_intrinsics = set([c for c in valid_cameras if intrinsics[c] is not None])
```

Comment:
Will we run into a scenario where we have a camera in `valid_cameras`, but not have intrinsics for it?

Reply:
We might not need to do this here and can just rely on `valid_cameras` input. Even if we do need to explicitly check for intrinsics, maybe do it before this function?

Reply:
yeah good point, I removed the intrinsics check. If intrinsics is missing, something is very wrong, lets just crash then.